pkg/util/client/client.go

```diff
@@ -35,32 +35,22 @@ func CreatePatch(before, after client.Object) (client.Patch, error) {
 // If strict is true, the resourceVersion will be part of the patch, make this call fail if
 // client.Object was changed.
 func Patch(ctx context.Context, c client.Client, obj client.Object, strict bool, update func() (bool, error)) error {
-	objOriginal := obj.DeepCopyObject().(client.Object)
-	if strict {
-		// Clearing ResourceVersion from the original object to make sure it is included in the generated patch.
-		objOriginal.SetResourceVersion("")
-	}
-	updated, err := update()
-	if err != nil || !updated {
-		return err
-	}
-	patch, err := CreatePatch(objOriginal, obj)
-	if err != nil {
-		return err
-	}
-	if err = c.Patch(ctx, obj, patch); err != nil {
-		return err
-	}
-	return nil
+	return patch(ctx, c, obj, strict, false, update)
 }
 
 // PatchStatus applies the merge patch of client.Object status.
 // The resourceVersion will be part of the patch, make this call fail if
 // client.Object was changed.
 func PatchStatus(ctx context.Context, c client.Client, obj client.Object, update func() (bool, error)) error {
+	return patch(ctx, c, obj, true, true, update)
+}
+
+func patch(ctx context.Context, c client.Client, obj client.Object, strict bool, status bool, update func() (bool, error)) error {
```

Comment:
are there any other callers besides PatchStatus?

Reply:
Yes, "Patch".

Reply:
> What about decomposition like this?

It is an alternative but seems like more code.

>  I'm not very happy about these double booleans in signature

This is internal non-exported function, so I'm not too worried about it, but we can rework, for sure.
A common pattern in golang is to introduce a variadic list of "options" like for example cmp.Diff or client.List does, but it feels like overkill here.